test/e2e/functional_test.go

```diff
@@ -226,18 +226,21 @@ func (s *FunctionalSuite) TestEventOnNodeFail() {
 		}).
 		ExpectAuditEvents(
 			fixtures.HasInvolvedObject(workflow.WorkflowKind, uid),
-			func(t *testing.T, e corev1.Event) {
-				assert.Equal(t, "WorkflowRunning", e.Reason)
-			},
-			func(t *testing.T, e corev1.Event) {
-				assert.Equal(t, "WorkflowNodeFailed", e.Reason)
-				assert.Contains(t, e.Message, "Failed node failed-step-event-")
-				assert.Equal(t, e.Annotations["workflows.argoproj.io/node-type"], "Pod")
-				assert.Contains(t, e.Annotations["workflows.argoproj.io/node-name"], "failed-step-event-")
-			},
-			func(t *testing.T, e corev1.Event) {
-				assert.Equal(t, "WorkflowFailed", e.Reason)
-				assert.Equal(t, "failed with exit code 1", e.Message)
+			2,
+			func(t *testing.T, es []corev1.Event) {
+				for _, e := range es {
```

Comment:
these now appear out of order - which I'm guessing is something to do with Goroutine scheduling